### ISSUE TITLE
Api_dump is not decoding shader stage flags correctly

### ISSUE THREAD

**Issue by Tony-LunarG** [LunarG/VulkanTools#428]
From a dump of Hologram:
```
Thread 0, Frame 0:
vkCreateDescriptorSetLayout(device, pCreateInfo, pAllocator, pSetLayout) returns VkResult VK_SUCCESS (0):
    device:                         VkDevice = 0000026D28DBCB30
    pCreateInfo:                    const VkDescriptorSetLayoutCreateInfo* = 0000026D28BB7010:
        sType:                          VkStructureType = VK_STRUCTURE_TYPE_DESCRIPTOR_SET_LAYOUT_CREATE_INFO (32)
        pNext:                          const void* = NULL
        flags:                          VkDescriptorSetLayoutCreateFlags = 0
        bindingCount:                   uint32_t = 1
        pBindings:                      const VkDescriptorSetLayoutBinding* = 0000026D28BB0328
            pBindings[0]:                   const VkDescriptorSetLayoutBinding = 0000026D28BB0328:
                binding:                        uint32_t = 0
                descriptorType:                 VkDescriptorType = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER_DYNAMIC (9)
                descriptorCount:                uint32_t = 1
                stageFlags:                     VkShaderStageFlags = 1 (VK_SHADER_STAGE_VERTEX_BIT | VK_SHADER_STAGE_ALL_GRAPHICS | VK_SHADER_STAGE_ALL)
                pImmutableSamplers:             const VkSampler* = UNUSED
    pAllocator:                     const VkAllocationCallbacks* = NULL
    pSetLayout:                     VkDescriptorSetLayout* = 0000026D28D27500
```
Note that VkShaderStageFlags is set to the value 1, but the interpretation from api_dump suggests that not only is VK_SHADER_STAGE_VERTEX_BIT set, it implies that VK_SHADER_STAGE_ALL_GRAPHICS and VK_SHADER_STAGE_ALL are also set, which they clearly are not

**Comment by charles-lunarg** [comment]
Issue is that when dumping the VkShaderStageFlagBits bitmask, VK_SHADER_STAGE_ALL_GRAPHICS are not unique flags, but cover a range (0x1F and 0x7FFFFFFF). Fix: either add special case for this or make api_dump catch bitmasks that have non-unique flags and adjust accordingly. If there are no other instances in the spec of this occurring, the first option is much more attractive to implement.

**Comment by charles-lunarg** [comment]
Fixed in PR #840 
Found several more instances of this issue occurring and devised a code-gen solution to it. 
Identifies any non-unique bitmask value and in the code-gen uses an equality comparison instead of bitwise comparison.
